interact.py

```diff
@@ -76,6 +76,9 @@ def sample_sequence(personality, history, tokenizer, model, args, current_output
         prev = torch.topk(probs, 1)[1] if args.no_sample else torch.multinomial(probs, 1)
         if i < args.min_length and prev.item() in special_tokens_ids:
             while prev.item() in special_tokens_ids:
+                if probs.max().item() == 1:
```

Comment:
Isn't this logic assuming that you obtained `prev` by performing argmax?

If `args.no_sample` were `False`, your `prev` wouldn't necessarily be the one with the highest probability. So the warning you print here will be a false one in that scenario, right?

Reply:
if there is a probability of 1 on a token, `prev` will always be that token, regardless of `args.no_sample`.